docs/asdf/developer_api.rst

```diff
@@ -7,6 +7,11 @@ Developer API
 The classes and functions documented here will be of use to developers who wish
 to create their own custom ASDF types and extensions.
 
+.. automodapi:: asdf.asdf
+    :no-inheritance-diagram:
+    :include: SerializationContext
+    :skip: AsdfConversionWarning, AsdfDeprecationWarning, AsdfObject, AsdfSearchResult, AsdfWarning, DelimiterNotFoundError, Extension, ExtensionMetadata, ExtensionProxy, HistoryEntry, Software, ValidationError, Version
```

Comment:
automodapi does not appear to pass `:include:` to automodsumm when generating class tables. This resulted in many warnings as the table contained rows linking to undocumented classes. I don't know why it also wants to include everything imported but this ugly list of skips appears to work.

Reply:
Why don't you just specify an `__all__`. automodapi generates documentation via a `*` import of the module. If no `__all__` is defined for the module, then anything not prefixed with `_` will be imported and automodapi will try to generate documentation for it. See #1577

Reply:
Thanks! That explains why `:include:` did not do what I expected based on the documentation. Rather than commit to an __all__ for `asdf.asdf` (which seems more like it should be private) I moved `SerializationContext` and exposed it at `asdf.extension.SerializationContext` as it's public use appears to be exclusively for extension developers.